tests/rag_answers/test_evaluate.py

```diff
@@ -155,3 +157,78 @@ def test_evaluate_and_output_results_copes_with_empty_data(
     evaluate_and_output_results(mock_project_root, file_path, mock_evaluation_config)
 
     assert "There is no data to evaluate" in caplog.text
+
+
+def test_evaluate_and_output_results_writes_deepeval_test_run(
+    tmp_path,
+    mock_input_data,
+    mock_evaluation_config,
+    mock_run_deepeval_evaluation,
+    mocker,
+):
+    class DummyTestRun:
+        def __init__(self) -> None:
+            self.call_kwargs: dict[str, bool] | None = None
+
+        def model_dump(self, *, by_alias: bool, exclude_none: bool) -> dict[str, str]:
+            self.call_kwargs = {"by_alias": by_alias, "exclude_none": exclude_none}
+            return {"id": "test-run-123"}
+
+    test_run = DummyTestRun()
+    mocker.patch(
+        "govuk_chat_evaluation.rag_answers.evaluate.global_test_run_manager.get_test_run",
+        return_value=test_run,
+    )
+
+    evaluate_and_output_results(tmp_path, mock_input_data, mock_evaluation_config)
+
+    output_path = tmp_path / "deepeval_test_run.json"
+    assert output_path.exists()
+
+    with output_path.open() as f:
+        assert json.load(f) == {"id": "test-run-123"}
+
+    assert test_run.call_kwargs == {"by_alias": True, "exclude_none": True}
+
+
+def test_log_metric_errors_emits_warnings(caplog):
```

Comment:
Rather than unit testing this internal function we should have a test_evaluate_and_output_results_logs_metric_errors where we test that the public method has the side effects we expect from the function